enumerable_methods.rb

```diff
@@ -0,0 +1,224 @@
+# rubocop:disable all 
+# 2
+module Enumerable
+  # 3
+  def my_each
+    if block_given?
+      iterator = 0
+      arr = self.to_a
+      while iterator < arr.length
+        yield(arr[iterator])
+        iterator += 1
+      end
+      return arr
+    else
+      to_enum(__method__)
+    end
+  end
+
+  # 4
+  def my_each_with_index
+    if block_given?
+      iterator = 0
+      arr = self.to_a
+      while iterator < arr.length
+        yield(arr[iterator], iterator)
+        iterator += 1
+      end
+      return arr
```

Comment:
#my_each_with_index when a block is given when self is a range returns the range itself after calling the given block once for each element in self

Again, you have to return the range itself if the self is a range. Instead, you are returning an array.